### PR TITLE
Spec updates for 0.2.0

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/spec/beaconstate.nim [hunk 1]
```diff
@@ -315,25 +315,26 @@ func update_validator_registry*(state: var BeaconState) =
...
     state: BeaconState, attestation: Attestation, flags: UpdateFlags): bool =
   ## Check that an attestation follows the rules of being included in the state
   ## at the current slot. When acting as a proposer, the same rules need to
   ## be followed!
-  ##
-  ## https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#attestations-1
 
-  if not (attestation.data.slot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot):
+  # Can't underflow, because GENESIS_SLOT > MIN_ATTESTATION_INCLUSION_DELAY
+  if not (attestation.data.slot <= state.slot - MIN_ATTESTATION_INCLUSION_DELAY):
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: I guess we'll see those comments all over the codebase :/

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -358,28 +359,62 @@ proc checkAttestation*(
...
+    filterIt(get_crosslink_committees_at_slot(state, attestation.data.slot),
+             it.shard == attestation.data.shard),
+    it.committee)[0]
+
+  assert allIt(0 ..< len(crosslink_committee),
+               if get_bitfield_bit(
+                   attestation.aggregation_bitfield, it) == 0b0:
+                 get_bitfield_bit(attestation.custody_bitfield, it) == 0b0
+               else:
+                 true)
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: I'm not super fan of the indentation here

Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: It avoids 80-length lines

Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: Changed to be more like other indentation

**Pull request merged**